icefoganalytics/travel-authorization
# Add Status/Audit Table To Travel Authorization Details Read/Edit Views and Expense Edit View

Fixes https://github.com/icefoganalytics/travel-authorization/issues/141

Relates to:
- [Analysis, Expense Pages, 2023-12-19](https://docs.google.com/document/d/1cWEA-0w9ro1TyajfnffkYaKL42yYnA89i9gdQysht_A/edit?pli=1#heading=h.zc2nwn7nmjlu)

# Context

The bottom of the Travel Authorization -> Details tab is missing this table. 
The same table needs to exist on the Travel Authorization -> Expense tab so the component should be generic.

This component will track data about the initial form submission, approval, and any subsequent actions performed on this page, or the manager view of this page. Tweak the travel_authorization_action_logs table so that it can track the appropriate information.

![Image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/9a231ae2-f292-4ad5-9bdd-670fa1891479)

# Implementation

Add info to distinguish assignee from actor to travel authorization action logs table.
Add "/submit" endpoint so travel authorization state updates are now through stateful services.
Add appropriate logging to all travel authorization state changes.
Add ensure user service to create supervisors in the system when they are assigned actions.
Add "use" minimal support to "i18n" plugin.

# Screenshots

Expense Edit tab
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/e62016a4-7976-4ca5-b04b-2411db552db4)

Details Read tab after submission and expense claim submission
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/081a9fc6-2f81-4b72-9c8d-da435efa12b9)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to "My Travel Requests" via the top drop down nav.
5. Create a new "Travel Authorization" via the call-to-action button.
6. Scroll to the button of the page to see the action table.
7. Fill in the Details tab and generate an estimate.
8. Submit the travel authorization to a random user that exists. e.g. Daniel.Test@yukon.ca
9. After submission completes, scroll to the bottom of the pages to see the status table filled with some status information.
    > Note that the user link only works for the current user. I think the page is missing?
    > Note that the user has been created in the database if they didn't exist.
10. Note that the users listed in the table are clickable and have information loaded from the government directory.
11. Approve the travel request via the database console.
12. Go back to the app and refresh the page, you can now go to the Expense tab.
13. Prefill your expenses and add receipts for all the expenses.
14. Add some general ledger coding information.
15. Submit the expense sheet to your supervisor.
    > Note that the expense read view has not been built.
16. Go back to the Details tab and scroll to the bottom, note that the status table has more info in it now.
